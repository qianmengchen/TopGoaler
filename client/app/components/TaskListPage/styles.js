import { StyleSheet } from 'react-native';

const list = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'stretch'
  },
  title: {
    textAlign: 'center',
    fontSize: 20,
    fontWeight: 'bold',
    marginTop: 10
  }
});

export { list };
