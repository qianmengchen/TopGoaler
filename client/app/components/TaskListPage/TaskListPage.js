import React, { Component } from 'react';
import { ScrollView, Text } from 'react-native';
import { TaskDetail } from '../TaskDetail/index';
import { list } from './styles';

class TaskListPage extends Component {
  _map_period_to_string = raw => {
    const period_map = { 0: 'day', 1: 'week', 2: 'month' };
    return period_map[raw];
  };

  _check_is_task_enrolled = tid => {
    return this.props.user_task_ids.includes(tid);
  };

  render() {
    return (
      <ScrollView>
        <Text style={list.title}>Channel Tasks</Text>
        {this.props.tasks.map(task => (
          <TaskDetail
            key={task.task_id}
            task_id={task.task_id}
            enrolled={this._check_is_task_enrolled(task.task_id)}
            title={task.title}
            period={this._map_period_to_string(task.period)}
            pattern={task.pattern}
            points={task.point}
          />
        ))}
      </ScrollView>
    );
  }
}

export default TaskListPage;
