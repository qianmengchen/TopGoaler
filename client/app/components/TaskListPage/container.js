import { connect } from 'react-redux';
import TaskListPage from './TaskListPage';

const mapStateToProps = () => ({
  // TODO
});

export default connect(mapStateToProps)(TaskListPage);
