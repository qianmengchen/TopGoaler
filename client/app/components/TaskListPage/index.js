import TaskListPage from './container';

export { TaskListPage };
