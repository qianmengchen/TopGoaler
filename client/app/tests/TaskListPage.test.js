import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunkMiddleware from 'redux-thunk';
require('isomorphic-fetch');

import TaskListPage from '../components/TaskListPage/TaskListPage';
const middlewares = [thunkMiddleware]; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

const initialTaskState = {
  tasks: ['do frontend', 'do backend', 'do MVP ftw'],
  isFetching: false,
  currentTaskPage: 'ChannelListPage'
};

describe('Testing Task List Page', () => {
  const tasks = [
    {
      id: 1,
      title: 'Review course material',
      channel_id: 1,
      point: 1,
      period: 1,
      pattern: 1
    }
  ];
  const mockInclude = { includes: jest.fn() };
  const wrapper = shallow(
    <TaskListPage tasks={tasks} user_task_ids={mockInclude} />,
    {
      context: { store: mockStore(initialTaskState) }
    }
  );
  const render = wrapper.dive();

  it('renders as expected', () => {
    expect(render).toMatchSnapshot();
  });

  it('should contain a Text for title', () => {
    expect(wrapper.find('Text')).toHaveLength(1);
  });
});
