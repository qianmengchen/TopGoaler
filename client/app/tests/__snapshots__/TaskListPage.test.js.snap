// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Task List Page renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScrollViewMock>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
          "marginTop": 10,
          "textAlign": "center",
        }
      }
    >
      Channel Tasks
    </Text>
    <Connect(TaskDetail)
      pattern={1}
      period="week"
      points={1}
      title="Review course material"
    />
  </ScrollViewMock>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [
    undefined,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
